src/librustc/hir/def_id.rs

```diff
@@ -252,6 +252,107 @@ impl DefId {
             format!("module `{}`", tcx.def_path_str(*self))
         }
     }
+
+    /// Check if a `DefId`'s path matches the given absolute type path usage.
+    // Uplifted from rust-lang/rust-clippy
+    pub fn match_path<'a, 'tcx>(self, tcx: TyCtxt<'a, 'tcx, 'tcx>, path: &[&str]) -> bool {
```

Comment:
I'm sorry for not noticing this until now, but such functionality doesn't belong on `DefId`, not even `TyCtxt` IMO.
Could we move this to the lint context, instead?

Reply:
Yes, I will do this tomorrow/Saturday!